clusterbuster

```diff
@@ -2427,6 +2430,86 @@ EOF
     done
 }
 
+function create_vm_deployment() {
+    local namespace=$1
+    local instance=$2
+    local -i replicas=$4
+    local -i replica=0
+    local depname="${namespace}-${workload}-${instance}"
+    while (( replica++ < replicas )) ; do
+        local name="${depname}-${replica}"
+        local vmname=$(mkpodname "$name")
+        create_object -n "$namespace" "$name" <<EOF
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  labels:
+    app: $name
+    kubevirt-vm: $vmname
+  name: $vmname
+$(indent 2 standard_deployment_metadata_yaml "$namespace" client)
+$(indent 2 standard_pod_metadata_yaml "$namespace" client)
+$(create_vm_spec -n "$vmname")
+EOF
+    done
+}
+
+function create_vm_spec() {
+    local image=${default_vm_image}
+    local vmname
+    local -i cores=1
+    local -i sockets=1
+    local -i threads=1
+    local mem=1024
+    local grace=30
+    local OPT
+    local OPTIND=1
+    local OPTARG
+    while getopts ':n:c:s:t:m:g:i:-' OPT; do
```

Comment:
The leading colon isn't necessary; getopts can handle this just fine.  The trailing - is only needed to handle long options; you're not doing that here, so omit that.

Reply:
Just made some big change, added global variables to the process_option for argument passing.